src/components/HomeLandingPage/HomeLandingPage.tsx

```diff
@@ -1,26 +1,38 @@
 import React, { useState } from 'react';
 import { useQuery } from 'react-apollo';
-import Typography from '@material-ui/core/Typography';
-import Grid from '@material-ui/core/Grid';
+import { Grid, Typography } from '@material-ui/core';
 import AccountCircleIcon from '@material-ui/icons/AccountCircle';
-
-import { useHistory } from 'react-router-dom';
+import { useHistory, useLocation } from 'react-router-dom';
 import MenuTabs from '../common/MenuTabs';
 import AddEventButton from './AddEventButton';
 import EventCard from './EventCard';
 import useAllEvents from '../../graphql/queries/hooks/events';
 import { Event, GET_ALL_EVENTS } from '../../graphql/queries/events';
 import '../../styles/HomeLandingPage.css';
 
+type LocationState = { addedEventId: string | null };
+
 const HomeLandingPage = () => {
+  const history = useHistory();
+  const location = useLocation<LocationState>();
```

Comment:
Can you explain why we use `useLocation<LocationState>();` just for my understanding? Why did you not use `useState()`?

Reply:
Sure, so `useState()` is used for component state. But I'm using `useLocation()` here to effectively pass `state` (which is part of `location`) between components upon url navigation changes.

Reply:
Let me know if you want me to go anymore in depth :)